### PR TITLE
Pass GH env vars to e2e test org created when possible

### PR DESCRIPTION
**Pull request by adrien-layer**

### CODE REVIEW

**Review by dpp23 — APPROVED** [review]

File: test/e2e/conftest.py [hunk 1]
```diff
@@ -250,10 +250,16 @@ def pseudo_random_project_name(fixture_request: Any) -> str:
...
+        "GITHUB_RUN_NUMBER"
+    )
+    random_suffix = (
+        f"{gh_run_id}-{gh_run_number}"
+        if gh_run_id
+        else str(uuid.uuid4()).replace("-", "")
+    )
+    random_suffix_that_fits = random_suffix[: name_max_length - len(name_prefix)]
+    name = f"{name_prefix}{random_suffix_that_fits}"
     display_name = f"SDK E2E Test Organization Account - {random_suffix}"
```
Comment (dpp23) [inline test/e2e/conftest.py:263]: Maybe putting it in the Display name is sufficient?

Comment (adrien-layer) [reply test/e2e/conftest.py:263]: i thought we didnt see it in the table view in the admin panel but we do, will update

File: test/e2e/conftest.py [hunk 2]
```diff
@@ -250,11 +250,17 @@ def pseudo_random_project_name(fixture_request: Any) -> str:
...
-    name = f"{name_prefix}{random_suffix}"
-    display_name = f"SDK E2E Test Organization Account - {random_suffix}"
+    random_suffix = str(uuid.uuid4()).replace("-", "")
+    random_suffix_that_fits = random_suffix[: name_max_length - len(name_prefix)]
+    name = f"{name_prefix}{random_suffix_that_fits}"
+    display_name = "SDK E2E Test Organization Account"
+    gh_run_id = os.getenv("GITHUB_RUN_ID")
+    gh_run_attempt = os.getenv("GITHUB_RUN_ATTEMPT")
+    if gh_run_id:
+        display_name += f" - {gh_run_id}:{gh_run_attempt}"
```
Comment (dpp23) [inline test/e2e/conftest.py:260]: Can we still add the job name? will be useful to quickly point us to the 3.7 or 3.8 run within the workflow

Comment (adrien-layer) [reply test/e2e/conftest.py:260]: problem is it doesnt add the matrix variable
Screenshot 2022-09-05 at 12 18 25

Comment (dpp23) [reply test/e2e/conftest.py:260]: That is annoying. You can pass it as:

```
diff --git a/.github/workflows/check.yml b/.github/workflows/check.yml
index 8e83ef3..908ff64 100644
--- a/.github/workflows/check.yml
+++ b/.github/workflows/check.yml
@@ -95,6 +95,7 @@ jobs:
     env:
       DD_GIT_COMMIT_AUTHOR_EMAIL: ci@layer.ai
       DD_GIT_COMMIT_COMMITTER_EMAIL: ci@layer.ai
+      GITHUB_JOB_PYTHON_VERSION: ${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v2
       - run: git fetch origin ${{ github.event.pull_request.base.sha }}
diff --git a/.github/workflows/release.yml b/.github/workflows/release.yml
index ff10d97..5431fe2 100644
--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -73,6 +73,7 @@ jobs:
     env:
       DD_GIT_COMMIT_AUTHOR_EMAIL: ci@layer.ai
       DD_GIT_COMMIT_COMMITTER_EMAIL: ci@layer.ai
+      GITHUB_JOB_PYTHON_VERSION: ${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v2
       - run: git fetch origin ${{ github.event.pull_request.base.sha }}
```

Comment (adrien-layer) [reply test/e2e/conftest.py]: this var seems empty for me

Screenshot 2022-09-05 at 13 16 49

**Pull request merged**